homeassistant/components/vorwerk/__init__.py

```diff
@@ -0,0 +1,109 @@
+"""Support for botvac connected Vorwerk vacuum cleaners."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pybotvac.exceptions import NeatoException
+from pybotvac.robot import Robot
+from pybotvac.vorwerk import Vorwerk
+
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.exceptions import ConfigEntryNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+from homeassistant.util import Throttle
+
+from .const import (
+    VORWERK_DOMAIN, 
+    VORWERK_ROBOTS,
+    VORWERK_ROBOT_ENDPOINT, 
+    VORWERK_ROBOT_NAME, 
+    VORWERK_ROBOT_SECRET, 
+    VORWERK_ROBOT_SERIAL, VORWERK_ROBOT_TRAITS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+VORWERK_SCHEMA = vol.Schema(
+    vol.All(
+        {
+            vol.Required(VORWERK_ROBOT_NAME): cv.string,
+            vol.Required(VORWERK_ROBOT_SERIAL): cv.string,
+            vol.Required(VORWERK_ROBOT_SECRET): cv.string,
+            vol.Optional(VORWERK_ROBOT_ENDPOINT, default="https://nucleo.ksecosys.com:4443"): cv.string,
+        }
+    )
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {VORWERK_DOMAIN: vol.Schema(vol.All(cv.ensure_list, [VORWERK_SCHEMA]))},
+     extra=vol.ALLOW_EXTRA
+)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
+    """Set up the Vorwerk component."""
+    hass.data[VORWERK_DOMAIN] = {}
+
+    if VORWERK_DOMAIN in config:
```

Comment:
Since this is a new component, importing YAML config is not needed

Reply:
But it is useful to configure the robot just by serial and secret. 
In the config flow it queries the vorwerk cloud to get this informations.

For the normal end user the serial and secret way is overkill but it more stable.

So my Idea was to have the simple way in the UI and the technical way in the config.

Reply:
For non developers it isn't that easy to get serial and secret. I think it's better to use the same workflow as the app.